lib/ansible/modules/files/replace.py

```diff
@@ -224,7 +224,11 @@ def main():
         module.fail_json(rc=257, msg='Path %s does not exist !' % path)
     else:
         f = open(path, 'rb')
-        contents = to_text(f.read(), errors='surrogate_or_strict')
+        data = f.read()
+        try:
+            contents = to_native(data, errors='surrogate_or_strict')
+        except:
+            contents = bytes(data)
```

Comment:
The file is opened in binary mode.  So there's no reason to try converting here as data is already bytes.